src/taipy/core/config/data_node_config.py

```diff
@@ -267,15 +296,26 @@ def _update(self, as_dict, default_section=None):
         self._storage_type = as_dict.pop(self._STORAGE_TYPE_KEY, self._storage_type)
         if self._storage_type is None and default_section:
             self._storage_type = default_section.storage_type
+
         self._scope = as_dict.pop(self._SCOPE_KEY, self._scope)
         if self._scope is None and default_section:
-            self._scope = default_section.scope
+            if default_section.scope and self._storage_type == default_section.storage_type:
+                self._scope = default_section.scope
+            else:
+                self._scope = self._DEFAULT_SCOPE
```

Comment:
this is great! but I strongly believes we should document it carefully as it can very easily become an "unexpected behavior" on the user side without detailed explanation in the doc

Reply:
Yes. I agree.